### PR TITLE
SP function

### PR DESCRIPTION
**Pull request by divijsharma18**

### CODE REVIEW

File: desidiff/nb/desidiff.ipynb [hunk 1]
```diff
@@ -196,15 +211,66 @@
...
+    "            for b in difflux.keys():\n",
+    "                wavelengths += difwave[b].tolist()\n",
+    "            \n",
+    "            flux = []\n",
+    "            #Combine the differenced flux into one \n",
+    "            for b in difflux.keys():\n",
+    "                flux += difflux[b].tolist()\n",
+    "            \n",
+    "            #Code to make wavelengths monotonic with corresponding changes in flux:\n",
+    "            wavelengths, flux = (list(t) for t in zip(*sorted(zip(wavelengths, flux))))\n",
```
Comment (AlexGKim) [inline desidiff/nb/desidiff.ipynb:226]: I am not familiar with this syntax, in particular, what happens in the sorting of a zip. We want to sort over wavelengths, the following is straightforward.

```
as = numpy.argsort(wavelengths)
wavelengths=wavelengths[as]
flux=flux[as]
```

**Review by AlexGKim — APPROVED** [review]
Looks good though look at the one comment, which I recommend gets changed since the code meaning is opaque to me.

File: desidiff/nb/desidiff.ipynb [hunk 2]
```diff
@@ -110,12 +122,15 @@
...
-    "        #Stores a tuple = (TARGETID, Z, ZERR,ZWARN, SPECTYPE)\n",
+    "        #Stores a tuple = (TARGETID, Z, ZERR,ZWARN, SPECTYPE, RA, DEC)\n",
     "        z_data = []\n",
+    "        ra_dec_data = []\n",
     "        for t in tid:\n",
     "            tid_ind = list(z['TARGETID']).index(t)\n",
+    "            ra_dec_ind = list(ra_dec['TARGETID']).index(t)\n",
     "            z_data.append((tid, z[tid_ind]['Z'], z[tid_ind]['ZERR'], z[tid_ind]['ZWARN'], z[tid_ind]['SPECTYPE']))\n",
-    "        \n",
+    "            ra_dec_data.append((tid, ra_dec[ra_dec_ind]['TARGET_RA'], ra_dec[ra_dec_ind]['TARGET_DEC']))\n",
```
Comment (AlexGKim) [inline desidiff/nb/desidiff.ipynb:132]: Should this be t and not tid?

**Review by AlexGKim — COMMENTED** [review]
I think there is a bug with tid

**Comment by divijsharma18** [comment]
I was actually thinking about that too. I don't need to append tid at
all actually, but I just wrote it because that is what we have done
for z_data. Should the z_data line also say t instead of tid?

**Comment by divijsharma18** [comment]
I see we have changed that in the master recently

**Pull request closed without merging**